src/Apps/Artist/Components/ArtistHeader.tsx

```diff
@@ -106,25 +134,39 @@ export class LargeArtistHeader extends Component<Props> {
       <HorizontalPadding>
         <Box width="100%">
           {hasImages && (
-            <Carousel
-              height="200px"
-              options={{ pageDots: false }}
-              data={carousel.images as object[]}
-              render={(slide: Image, slideIndex: number) => {
-                return (
-                  <a href={slide.href} onClick={() => this.onClickSlide(slide)}>
-                    <Image
-                      px={0.3}
-                      lazyLoad={slideIndex > 5}
-                      src={slide.resized.url}
-                      width={slide.resized.width}
-                      height={slide.resized.height}
-                      preventRightClick={!isAdmin}
-                    />
-                  </a>
-                )
-              }}
-            />
+            <>
+              <Carousel
+                height="200px"
+                options={{ pageDots: false }}
+                data={carousel.images as object[]}
+                render={(slide: Image, slideIndex: number) => {
+                  return (
+                    <a
+                      href={slide.href}
+                      onClick={() => this.onClickSlide(slide)}
+                    >
+                      <Image
+                        px={0.3}
+                        lazyLoad={slideIndex > 5}
+                        src={slide.resized.url}
+                        width={slide.resized.width}
+                        height={slide.resized.height}
+                        preventRightClick={!isAdmin}
+                      />
+                    </a>
+                  )
+                }}
+              />
+              <WorksForSaleButton>
+                <Button
+                  variant="primaryWhite"
+                  size="small"
+                  onClick={() => this.handleBrowseWorksClick(props.artist.slug)}
```

Comment:
This came up with client side routing (and we were discussing with @joeyAghion in lunch and learn), but we should try to avoid using custom click handlers, to handle navigation and routing when possible.

This feels like a link to the /works-for-sale tab of an artist page, yet uses `window.location.assign` in an `onClick` (which will break client-side routing), rather than an anchor tag with `href` pointing to that (for users w/o javascript).

Reply:
Oh!

I did this so that we could get custom tracking around the button. Is there a solution that satisfies both of these needs?

Reply:
You can still have a no-op `onClick` just for analytics tracking purposes, but the component should probably just be a [styled] anchor tag with `href`.